src/test/java/com/google/sps/TripTest.java

```diff
@@ -69,6 +97,36 @@ public void testTripConstructorSingleDay() {
     Assert.assertEquals(trip.getNumDays(), 1);
   }
 
+  // Test the trip build entity
+  @Test
+  public void testTripBuildEntity() {
+    Entity tripEntity = Trip.buildEntity(TRIP_NAME, DESTINATION_NAME, IMAGE_SRC, 
+      START_DATE_STRING, END_DATE_STRING, KeyFactory.createKey(PARENT_KEY_KIND, PARENT_KEY_NAME));
+
+    Assert.assertEquals(TRIP_NAME, tripEntity.getProperty(Trip.TRIP_NAME));
+    Assert.assertEquals(DESTINATION_NAME, tripEntity.getProperty(Trip.DESTINATION_NAME));
+    Assert.assertEquals(IMAGE_SRC, tripEntity.getProperty(Trip.IMAGE_SRC));
+    Assert.assertEquals(START_DATE_STRING, tripEntity.getProperty(Trip.START_DATE));
+    Assert.assertEquals(END_DATE_STRING, tripEntity.getProperty(Trip.END_DATE));
+  }
+
+  // Test the trip build trip from entity
+  @Test
+  public void testBuildTripFromEntity() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity tripEntity = Trip.buildEntity(TRIP_NAME, DESTINATION_NAME, IMAGE_SRC, 
+      START_DATE_STRING, END_DATE_STRING, KeyFactory.createKey(PARENT_KEY_KIND, 
+      PARENT_KEY_NAME));
+    datastore.put(tripEntity);
+    Trip trip = Trip.buildTripFromEntity(tripEntity);
+
+    Assert.assertEquals(trip.getTripName(), TRIP_NAME);
+    Assert.assertEquals(trip.getDestinationName(), DESTINATION_NAME);
+    Assert.assertEquals(trip.getImageSrc(), IMAGE_SRC);
+    Assert.assertEquals(trip.getStartDate(), START_DATE);
+    Assert.assertEquals(trip.getEndDate(), END_DATE);
+  }
+
```

Comment:
This is probably how I should have tested my event stuff as well but I actually just converted in from object --> entity --> object and then confirmed that those were the same. No change required just kind of thinking out loud.

Reply:
Ok, gotcha 👍